app/client/src/widgets/TableWidgetV2/widget/derived.js

```diff
@@ -317,6 +317,48 @@ export default {
     const sortByColumnId = props.sortOrder.column;
 
     let sortedTableData;
+    /* 
+    Check if there are select columns, 
+    and if the columns are sorting by label instead of default value 
+    */
+    const selectColumnKeysWithSortByLabel = [];
+    Object.keys(primaryColumns).forEach((id) => {
+      if (
+        primaryColumns[id] &&
+        primaryColumns[id].columnType === "select" &&
+        primaryColumns[id].sortBy &&
+        primaryColumns[id].sortBy === "label" &&
+        primaryColumns[id].selectOptions &&
+        primaryColumns[id].selectOptions.length
```

Comment:
```suggestion
        primaryColumns[id].columnType === "select" &&
        primaryColumns[id]?.sortBy === "label" &&
        primaryColumns[id]?.selectOptions?.length
```
Would this suffice? It will make this condition more readable.
Also for such big conditions it is better to store this in a readable variable(eg `isSortingByLabelForColumnID`)